crates/matrix-sdk-common/src/linked_chunk/lazy_loader.rs

```diff
@@ -196,6 +196,59 @@ where
     Ok(())
 }
 
+/// A pretty inefficient, test-only, function to rebuild a full `LinkedChunk`.
+#[doc(hidden)]
+pub fn from_all_chunks<const CAP: usize, Item, Gap>(
+    mut chunks: Vec<RawChunk<Item, Gap>>,
+) -> Result<Option<LinkedChunk<CAP, Item, Gap>>, LazyLoaderError>
+where
+    Item: Clone,
+    Gap: Clone,
+{
+    if chunks.is_empty() {
+        return Ok(None);
+    }
+
+    // Sort by `next` so that the search for the next chunk is faster (it should
+    // come first). The chunk with the biggest next chunk identifier comes first.
+    // Chunk with no next chunk comes last.
+    chunks.sort_by(|a, b| b.next.cmp(&a.next));
+
+    let last_chunk = chunks.pop().unwrap();
+    let last_chunk_identifier = last_chunk.identifier;
+    let chunk_identifier_generator =
+        ChunkIdentifierGenerator::new_from_previous_chunk_identifier(last_chunk_identifier);
+
+    let mut linked_chunk = from_last_chunk(Some(last_chunk), chunk_identifier_generator)?.unwrap();
```

Comment:
Not sure if you can get rid of this one, but at least put an `expect()` there instead.

Reply:
Sure, let's add an `expect`. For the record, it's purely used for testing purposes, but I share your concerns.

Reply:
I've changed for a `else { return Ok(None); }` here.